hera_qm/utils.py

```diff
@@ -610,3 +611,118 @@ def strip_extension(path, return_ext=False):
+        altitude should be in meters.
+        Default is None.
+        If None, will determine location from uv.telescope_name
+    telescope_name : str, optional
+        string with name of telescope.
+        Default is None. If None, will use uv.telescope_name
+    ant_indices_only : bool
+        If True, ignore polarizations and flag entire antennas when they appear, e.g. (1, 'Jee') --> 1.
+    by_ant_pol : bool
+        If True, expand all integer antenna indices into per-antpol entries using ant_pols
+    ant_pols : list of str
+        List of antenna polarizations strings e.g. 'Jee'. If not empty, strings in
+        the YAML must be in here or an error is raised. Required if by_ant_pol is True.
+
+    Returns
+    -------
+        uv : UVData or UVCal object
+            input uvdata / uvcal but now with flags applied
+    """
+    # check that uv is UVData or UVCal
+    if not issubclass(uv.__class__, (UVData, UVCal)):
+        raise NotImplementedError("uv must be a UVData or UVCal object.")
+    # if UVCal provided, get lst_array from times.
+    # If lat_lon_alt is not specified, try to infer it from the telescope name, which calfits files generally carry around
+    if not hasattr(uv, 'lst_array'):
+        if lat_lon_alt_degrees is None:
+            if telescope_name is None:
+                telescope_name = uv.telescope_name
+            if telescope_name.upper() in KNOWN_TELESCOPES:
+                lat = KNOWN_TELESCOPES[telescope_name.upper()]['latitude'] * 180 / np.pi
+                lon = KNOWN_TELESCOPES[telescope_name.upper()]['longitude'] * 180 / np.pi
+                alt = KNOWN_TELESCOPES[telescope_name.upper()]['altitude']
+                lat_lon_alt_degrees = np.asarray([lat, lon, alt])
+            else:
+                raise NotImplementedError(f'No known position for telescope {telescope_name}. lat_lon_alt_degrees must be specified.')
+
+        # calculate LST grid in hours from time grid and lat_lon_alt
+        lst_array = uvutils.get_lst_for_time(uv.time_array, *lat_lon_alt_degrees) * 12 / np.pi
+    else:
+        lst_array = np.unique(uv.lst_array) * 12  / np.pi
+
+    time_array = np.unique(uv.time_array)
+    # loop over spws to apply frequency flags.
+    for spw in range(uv.Nspws):
+        flagged_channels = metrics_io.read_a_priori_chan_flags(a_priori_flag_yaml, freqs=uv.freq_array[spw])
+        if np.any(flagged_channels >= uv.Nfreqs):
+            warnings.warn("Flagged channels were provided that exceed the maximum channel index. These flags are being dropped!")
+        if np.any(flagged_channels < 0):
+            warnings.warn("Flagged channels were provided with a negative channel index. These flags are being dropped!")
+        flagged_channels = flagged_channels[(flagged_channels>=0) & (flagged_channels<=uv.Nfreqs)]
+        if issubclass(uv.__class__, UVData):
+            uv.flag_array[:, spw, flagged_channels, :] = True
+        elif issubclass(uv.__class__, UVCal):
```

Comment:
Since we're no longer enforcing UVData or UVCal above, we should raise an error if nothing is happening here.  

Also, do we want to support UVFlag too? It's a bit annoying, since UVFlag has 3 different modes, but one of them looks like UVData, one looks like UVCal, and one is just a waterfall.

Reply:
I did think it might be nice to add `UVFlag` but I think let's make it an issue for a different PR if the need arises.